### DISCUSSION TITLE
Would you recommend normy to reduce a very large payload data from the backed where...

### DISCUSSION THREAD

**Discussion by zomars** [trpc/trpc#5449] [Q&A]
Would you recommend normy to reduce a very large payload data from the backed where there's a lot of repeated data? Say there's a lot of nested objects that have almost the same kind of properties and it gets exponentially larger. I'm looking to "normalize" data returned from the backed and re-normalize it again in the frontend.

Would this library help me achieve this? Or is not the recommended usage?

**Comment by KATT** [comment]
I think using devalue as a transformer would help, it does some deduping. 

Superjson bloats payloads a lot but you could likely use normy together with superjson too

We also have worked on a library called tupleson to do json serializing/deserializing that can do deduping on identical objects, but that's not really production ready yet

**Reply by KATT** [reply]
Maybe doing like gzip compression in your api handler could work too

**Reply by a-eid** [reply]
does it make sense for trpc to do so automatically ?
